### PR TITLE
Adds missing lightgbm tuner metrics for the case of higher is better

### PR DESCRIPTION
**Pull request by mavillan**
## Motivation
Missing [higher_is_better] metrics in LightGBMTuner.

## Description of the changes

When instantiating a LightGBMTuner with metrics `auc_mu` or `average_precision` and an study with direction equal to `maximize` you get this error:

```
ValueError: Study direction is inconsistent with the metric auc_mu. Please set 'minimize' as the direction.
```

I just added the missing metrics to the `self.higher_is_better` method of `LightGBMTuner`

### CODE REVIEW

**Comment by hvy** [comment]
@toshihikoyanase, @keisuke-umezawa could you perhaps review this PR covering additional metrics names?

**Review by toshihikoyanase — COMMENTED** [review]
Thank you for your PR. The change looks good to me.

Could you update the corresponding test case, please?

https://github.com/optuna/optuna/blob/327af0d5acab329a929ebb17930448c7a42f82f1/tests/integration_tests/lightgbm_tuner_tests/test_optimize.py#L138-L151

**Comment by mavillan** [comment]
@toshihikoyanase I have added all the metrics (and its aliases) corresponding to higher_is_better and also updated the test.

File: optuna/integration/_lightgbm_tuner/optimize.py [hunk 1]
```diff
@@ -146,7 +146,21 @@ def _metric_with_eval_at(self, metric: str) -> str:
...
+            "rank_xendcg",
+            "xendcg",
+            "xe_ndcg",
+            "xe_ndcg_mart",
+            "xendcg_mart",
+            "map",
+            "mean_average_precision",
+            "average_precision",
+        ]
+        return metric_name in supported_higher_is_better
```
Comment (toshihikoyanase) [inline optuna/integration/_lightgbm_tuner/optimize.py:163]: Could you revert this change, please? The metrics that you added (i.e., `landmark`, `rank_xendcg`, `xendcg`, `xe_ndcg`, `xe_ndcg_mart`, `xendcg_mart`) are converted to `ndcg` by `_handling_alias_parameters`.

https://github.com/optuna/optuna/blob/6568ef921b58b94e0eb2d427e60c4c28d4c74ac0/optuna/integration/_lightgbm_tuner/alias.py#L30-L55

**Review by toshihikoyanase — CHANGES_REQUESTED** [review]
Thank you for adding the test. You kindly added some parameters to `higher_is_better`, but I think it has been covered by preprocessing of the metric name. Please revert the change.

File: optuna/integration/_lightgbm_tuner/optimize.py (hunk 1 above)
Comment (mavillan) [reply optuna/integration/_lightgbm_tuner/optimize.py]: Right, I didn't notice the handling of aliases. I fixed it in the last commit.

**Review by keisuke-umezawa — APPROVED** [review]
LGTM based on this reference: https://lightgbm.readthedocs.io/en/latest/Parameters.html#metric-parameters

Thank you for your contribution!

File: optuna/integration/_lightgbm_tuner/optimize.py (hunk 1 above)
Comment (toshihikoyanase) [reply optuna/integration/_lightgbm_tuner/optimize.py]: Thanks!

**Review by toshihikoyanase — APPROVED** [review]
LGTM! Thank you for your contribution.
